### PR TITLE
Add solution of Einstein's equations for a generic gauge plane wave

### PR DESCRIPTION
**Pull request by kidder**
## Proposed changes

This adds a generic dynamic solution of Einstein's equations.  It should be useful for testing AMR.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/GaugePlaneWave.hpp [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+ * \partial_\rho g_{\mu\nu} = H' l_\rho l_\mu l_\nu,
+ * \f}
+ *
+ * The 3+1 quantities are
+ * \f{align}{
+ * \alpha & = \left( 1 + H \omega^2 \right)^{-1/2},\\
+ * \beta^i & = \frac{-H \omega k^i}{1 + H \omega^2},\\
+ * \gamma_{ij} & = \delta_{ij} + H k_i k_j,\\
+ * \gamma & = 1 + H \omega^2,\\
+ * \gamma^{ij} & = 1 - \frac{H k^i k^j}{1 + H \omega^2},\\
```
Comment (wthrowe) [inline src/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/GaugePlaneWave.hpp]: `1` -> `\delta^{ij}`

File: tests/Unit/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/Test_GaugePlaneWave.cpp [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+  struct factory_creation
+      : tt::ConformsTo<Options::protocols::FactoryCreation> {
+    using factory_classes = tmpl::map<
+        tmpl::pair<MathFunction<1, Frame::Inertial>,
+                   tmpl::list<MathFunctions::PowX<1, Frame::Inertial>>>>;
+  };
+};
+
+template <size_t Dim>
+struct GaugePlaneWaveProxy : gr::Solutions::GaugePlaneWave<Dim> {
```
Comment (wthrowe) [inline tests/Unit/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/Test_GaugePlaneWave.cpp]: This appears to be the infrastructure for a python test, but there's no python test.  I don't know if you intended to write one or if this just got left in.

**Review by wthrowe — CHANGES_REQUESTED** [review]

**Review by wthrowe — APPROVED** [review]
Looks good.  Squash.

**Comment by kidder** [comment]
squashed

**Pull request merged**